mlflow/utils/__init__.py

```diff
@@ -6,12 +6,15 @@
 from itertools import islice
 from sys import version_info
 
-_logger = logging.getLogger(__name__)
-
+import pydantic
+from packaging.version import Version
 
 PYTHON_VERSION = f"{version_info.major}.{version_info.minor}.{version_info.micro}"
 
 
+IS_PYDANTIC_V2 = Version(pydantic.__version__).major >= 2
```

Comment:
not sure if this file is the best place for it, but i noticed we define some version of this constant in many places and i wanted to unify